addons/udes_stock_cron/models/stock_picking.py

```diff
@@ -21,6 +21,74 @@
 class StockPicking(models.Model):
     _inherit = "stock.picking"
 
+    def _find_unreservable_moves(self, batch_size=1000, return_fast=False):
+        """
+        Find moves for which there is insufficient stock.
+
+        The u_handle_partials flag on the picking type is respected: if it is
+        True, partially reservable lines will not be reported.
+        """
+        # This method is patterned on SaleOrder._find_unfulfillable_order_lines
+        # in udes_sale_stock.  Changes applied there should be considered for
+        # this method too.
+        Location = self.env["stock.location"]
+        Move = self.env["stock.move"]
+        Quant = self.env["stock.quant"]
+
+        # Create empty record sets for moves.
+        unreservable_moves = Move.browse()
+
+        # Get unreserved stock for each product in locations
+        locations = Location.get_available_stock_locations()
+        stock_for_products = defaultdict(int)
+        skip_states = ("assigned", "done", "cancel")
+
+        _logger.info(_("Checking reservability for %d pickings."), len(self))
+        for r, batch in self.batched(size=batch_size):
+            _logger.info("Checking pickings %d-%d", r[0], r[-1])
+            # Cache the needed fields and only the needed fields
+            # Caching too much is expensive here because of the sheer number of
+            # records processed: Odoo's field loading becomes a bottleneck and
+            # memory usage skyrockets.
+            # Caching too little is also expensive since:
+            #  * cache misses cause unused fields to be loaded in due to
+            #    prefetching, which leads to overcaching, described above
+            #  * cache misses for stock move fields result in hundreds of small
+            #    loads per batch, which is inefficient due to overheads
+            # For non-relational fields, read() is used instead of mapped()
+            # because it allows for loading of specific fields, whereas
+            # mapped() will load in as many fields as it can due to prefetching.
+            # with_context(prefetch_fields=False) could be used with mapped()
+            # but is limited to a single column at a time.
+            batch.with_context(prefetch_fields=False).mapped("move_lines")
+            batch.mapped("move_lines").read(
```

Comment:
Do we need the `mapped` here in udes 14?

Reply:
Mapped not needed for relational fields any more. We could use the variable mentioned above.